### PR TITLE
ECSRA-57 Swagger documentation

### PR DESCRIPTION
**Pull request by MariiaLobanova**
Added dependency in pom.xml, added needed endpoints in SecurityConfig class
Image 12 01 24 at 15 56
Image 12 01 24 at 15 54
Image 12 01 24 at 15 53

### CODE REVIEW

**Review by MenilV — CHANGES_REQUESTED** [review]
let's fix the other PR first as are some pieces of the same code in both PRs

File: pom.xml [hunk 1]
```diff
@@ -52,6 +52,54 @@
...
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-validation</artifactId>
+		</dependency>
+
+
+		<dependency>
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<scope>runtime</scope>
```
[the remark this answers is not in the document]
Comment (MariiaLobanova) [reply pom.xml]: No, dont need, deleted its dependency

File: src/main/java/startsteps/ECommerceShop/config/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                                    @NonNull FilterChain filterChain)
+            throws ServletException, IOException {
+        final String authHeader = request.getHeader("Authorization");
+        final String jwt;
+        final String userEmail;
+        if (StringUtils.isEmpty(authHeader) || !StringUtils.startsWith(authHeader, "Bearer")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+        jwt = authHeader.substring(7);
```
[the remark this answers is not in the document]
Comment (MariiaLobanova) [reply src/main/java/startsteps/ECommerceShop/config/JwtAuthenticationFilter.java]: I gave a comment in previous PR: In my case "Bearer " includes 7 characters, and then goes token-genereted code and it is extracts this part to use. Is not  it ok?

**Review by MenilV — APPROVED** [review]
awesome, approved

**Pull request merged**